### ISSUE TITLE
Speedreader crashes while opening OTR window

### ISSUE THREAD

**Issue by iefremov** [brave/brave-browser#9825]
STR: 
1) Run browser with enabled speedreader (either via brave://flags or `--enable-features=Speedreader`)
2) Open incognito or TOR window
3) browser crashes

**Comment by bsclifton** [comment]
Added missing milestone to this issue and https://github.com/brave/brave-core/pull/5584 👍

**Comment by kjozwiak** [comment]
@iefremov looks like it's still crashing. I'm not sure how often it happened before, but I'm still running into the following crashes from time to time whenever I start opening/closing `Tor` & `Private` windows using the following build:


```
Brave | 1.11.64 Chromium: 83.0.4103.97 (Official Build) dev (64-bit)
-- | --
Revision | 326d148b9655369b86498d9ecca39f63dd2bdd2d-refs/branch-heads/4103@{#657}
OS | macOS Version 10.15.5 (Build 19F101)
```

Crashes:

* Uploaded Crash Report ID: `3e6f0000-1c55-8704-0000-000000000000`
* Uploaded Crash Report ID: `3f6f0000-1c55-8704-0000-000000000000`
* Uploaded Crash Report ID: `4a6f0000-1c55-8704-0000-000000000000`

```
[ 00 ] extensions::BraveTorClientUpdater::SetExecutablePath(base::FilePath const&)
[ 01 ] void base::internal::ReplyAdapter<base::FilePath, base::FilePath const&>(base::OnceCallback<void (base::FilePath const&)>, std::__1::unique_ptr<base::FilePath, std::__1::default_delete<base::FilePath> >*)
[ 02 ] base::internal::Invoker<base::internal::BindState<void (*)(base::OnceCallback<void (base::FilePath const&)>, std::__1::unique_ptr<base::FilePath, std::__1::default_delete<base::FilePath> >*), base::OnceCallback<void (base::FilePath const&)>, base::internal::OwnedWrapper<std::__1::unique_ptr<base::FilePath, std::__1::default_delete<base::FilePath> >, std::__1::default_delete<std::__1::unique_ptr<base::FilePath, std::__1::default_delete<base::FilePath> > > > >, void ()>::RunOnce(base::internal::BindStateBase*)
[ 03 ] base::(anonymous namespace)::PostTaskAndReplyRelay::RunReply(base::(anonymous namespace)::PostTaskAndReplyRelay)
[ 04 ] base::internal::Invoker<base::internal::BindState<void (*)(base::(anonymous namespace)::PostTaskAndReplyRelay), base::(anonymous namespace)::PostTaskAndReplyRelay>, void ()>::RunOnce(base::internal::BindStateBase*)
[ 05 ] base::TaskAnnotator::RunTask(char const*, base::PendingTask*)
[ 06 ] base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWorkImpl(base::sequence_manager::LazyNow*, bool*)
[ 07 ] base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork()
[ 08 ] invocation function for block in base::MessagePumpCFRunLoopBase::RunWorkSource(void*)
[ 09 ] base::mac::CallWithEHFrame(void () block_pointer)
[ 10 ] base::MessagePumpCFRunLoopBase::RunWorkSource(void*)
[ 11 ] 0x7fff38a92de2
[ 12 ] 0x7fff38a92d81
[ 13 ] 0x7fff38a92b9b
[ 14 ] 0x7fff38a918ca
[ 15 ] 0x7fff38a90ece
[ 16 ] RunCurrentEventLoopInMode
[ 17 ] ReceiveNextEventCommon
[ 18 ] _BlockUntilNextEventMatchingListInModeWithFilter
[ 19 ] _DPSNextEvent
[ 20 ] -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:]
[ 21 ] __71-[BrowserCrApplication nextEventMatchingMask:untilDate:inMode:dequeue:]_block_invoke
[ 22 ] base::mac::CallWithEHFrame(void () block_pointer)
[ 23 ] -[BrowserCrApplication nextEventMatchingMask:untilDate:inMode:dequeue:]
[ 24 ] -[NSApplication run]
[ 25 ] base::MessagePumpNSApplication::DoRun(base::MessagePump::Delegate*)
[ 26 ] base::MessagePumpCFRunLoopBase::Run(base::MessagePump::Delegate*)
[ 27 ] base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run(bool, base::TimeDelta)
[ 28 ] base::RunLoop::Run()
[ 29 ] ChromeBrowserMainParts::MainMessageLoopRun(int*)
[ 30 ] content::BrowserMainLoop::RunMainMessageLoopParts()
[ 31 ] content::BrowserMainRunnerImpl::Run()
[ 32 ] content::BrowserMain(content::MainFunctionParams const&)
[ 33 ] content::ContentMainRunnerImpl::RunServiceManager(content::MainFunctionParams&, bool)
[ 34 ] content::ContentMainRunnerImpl::Run(bool)
[ 35 ] service_manager::Main(service_manager::MainParams const&)
[ 36 ] content::ContentMain(content::ContentMainParams const&)
[ 37 ] ChromeMain
[ 38 ] main
[ 39 ] start
```

**Comment by kjozwiak** [comment]
@rebron @bsclifton is this something we want in `1.11.x`? I'll create a follow up issue as there's already code that landed for this issue and label this as `QA/No` as there's no point in testing as it still crashes.

**Comment by kjozwiak** [comment]
Created https://github.com/brave/brave-browser/issues/10283 and labeled this as `QA/No` as speedreader isn't being released in `1.11.x` and is behind a flag as per the above.
